org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/AbstractMenuPanel.java

```diff
@@ -313,6 +315,9 @@ private void onNewRecord(Treeitem selectedItem) {
         {
 			int menuId = Integer.parseInt((String)selectedItem.getValue());
 			MMenu menu = new MMenu(Env.getCtx(), menuId, null);
+			IDesktop desktop = SessionManager.getAppDesktop();
+			if (desktop instanceof AbstractDesktop)
+				((AbstractDesktop)desktop).setPredefinedContextVariables(menu.getPredefinedContextVariables());

```

Comment:
how does the PredefinedContextVariables at AD_Window and AD_Menu suppose to works together ?

Reply:
The menu predefined variables are added to the context just when the window is opened from the menu, the window variables are added to the context always.
I have not specifically a use case for that - I made some tests for example making a window read-only when opened by zoom, but read-write with specific filters when opened from menu (as described in [the ticket](https://idempiere.atlassian.net/browse/IDEMPIERE-4713?focusedCommentId=46961))